Added get_options method

pkopparla opened the pull request:
<!-- Feel free to remove check-list items aren't relevant to your change -->

- [X] Closes #5698 
- [X] Passes `pre-commit run --all-files`
- [X] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [X] New functions/methods are listed in `api.rst`

Added a `get_options` method under `core/options.py`. At the moment it returns the OPTIONS dictionary as is to keep it very simple, if a formatted print statement with text is desirable I can try to add it.

Edit: Some tests are failing, but they are unrelated to my changes. Not sure how to address the errors.

Illviljan commented:
The returned dict should be read only, otherwise you can bypass all the validation checks that's done in `set_options`. I think you can use `FrozenDict` for that:
https://github.com/pydata/xarray/blob/56cc0778d5edcb0173425b5850b834457c790210/xarray/core/utils.py#L476
Not sure if FrozenDict propagates the typing though, once OPTIONS is typed.

We probably need to make sure that `get_options` is updated if `set_options` has been used. A test for that would be nice.

max-sixty commented:
This looks good., thanks @pkopparla  Thanks for reviewing @Illviljan .

Agree re a test — at least to just test it runs, albeit trivial. 

`FrozenDict` would be better, though I think fine to do in another PR. We could return `dict(OPTIONS)` if we're worried people will attempt to set invalid values.

pkopparla commented:
Thank you @Illviljan @max-sixty 
I've changed the return type to `FrozenDict` and added a check to make sure that `set_options` changes are reflected in `get_options`.

Illviljan commented on xarray/core/options.py:
```diff
@@ -269,3 +271,56 @@ def __enter__(self):
+
+    Currently supported options:
+
+    - ``display_width``: maximum display width for ``repr`` on xarray objects.
```
A new format for the parameters was just merged, #5702. Could you copy that one instead?

pkopparla replied:
Thanks for pointing that out, I've just copied it now.

max-sixty commented on doc/whats-new.rst:
```diff
@@ -22,6 +22,7 @@ v0.19.1 (unreleased)
 
 New Features
 ~~~~~~~~~~~~
+- Added a get_options method to Xarray (:issue:`5698`)
```
```suggestion
- Added a ``get_options`` method to xarray's root namespace (:issue:`5698`)
```

(There is also a way to link to the method itself but I'm on my phone!)

dcherian commented on doc/whats-new.rst:
```diff
@@ -22,6 +22,7 @@ v0.19.1 (unreleased)
 
 New Features
 ~~~~~~~~~~~~
+- Added a ``get_options`` method to xarray's root namespace (:issue:`5698`)
```
```suggestion
- Added a :py:func:`get_options` method to xarray's root namespace (:issue:`5698`, :pull:`5716`)
```

dcherian commented on xarray/core/options.py:
```diff
@@ -269,3 +271,73 @@ def __enter__(self):
+    """
+    Get options for xarray.
+
+    Parameters
```
Shall we instead just refer the user to the docstring for `set_options`. That way it is only ever described in one place?

pkopparla replied:
I've just done this. I also added an additional parameter to the test to make sure that `get_options` is actually tracking `set_options` changes and not just matching because of a default value or so.

dcherian commented on xarray/core/options.py:
```diff
@@ -269,3 +271,73 @@ def __enter__(self):
+
+        * ``True`` : to always expand data variables
+        * ``False`` : to always collapse data variables
+        * ``default`` : to expand unless over a pre-defined limit
```
```suggestion
        * ``default`` : to expand unless over a pre-defined limit
        
    See Also
    --------
    set_options
```

dcherian reviewed (commented):
Thanks @pkopparla This is a great improvement!

dcherian reviewed (approved):
Thanks @pkopparla 

I'll merge in a couple of days if there are no more comments

pkopparla commented:
Awesome, thanks @dcherian and all the reviewers!
